extensions/dicom-sr/src/getSopClassHandlerModule.js

```diff
@@ -93,22 +89,36 @@ function _getDisplaySetsFromSeries(
     SeriesInstanceUID,
     StudyInstanceUID,
     SOPClassHandlerId: `${id}.sopClassHandlerModule.${sopClassHandlerName}`,
-    referencedImages: _getReferencedImagesList(ContentSequence),
-    measurements: _getMeasurements(ContentSequence),
+    referencedImages: null,
+    measurements: null,
+    isDerivedDisplaySet: true,
+    isLoaded: false,
     sopClassUids,
+    instance,
   };
 
+  displaySet.load = () => _load(displaySet, servicesManager, extensionManager);
+
+  return [displaySet];
+}
+
+function _load(displaySet, servicesManager, extensionManager) {
+  const { DisplaySetService, MeasurementService } = servicesManager.services;
+  const dataSources = extensionManager.getDataSources();
+  const dataSource = dataSources[0];
+
+  const { ContentSequence } = displaySet.instance;
+
+  displaySet.referencedImages = _getReferencedImagesList(ContentSequence);
+  displaySet.measurements = _getMeasurements(ContentSequence);
+
   const mappings = MeasurementService.getSourceMappings(
     'CornerstoneTools',
     '4'
   );
 
-  if (isRehydratable(displaySet, mappings)) {
-    displaySet.isLocked = false;
-    displaySet.isHydrated = false;
-  } else {
-    displaySet.isLocked = true;
-  }
+  displaySet.isHydrated = false;
```

Comment:
Hydrated check is interesting. If we cache displaySets, do we need to clear this or displaySets on mode exit?